lib/ansible/module_utils/mysql.py

```diff
@@ -61,6 +61,17 @@ def mysql_connect(module, login_user=None, login_password=None, config_file='',
     if connect_timeout is not None:
         config['connect_timeout'] = connect_timeout
 
+    # If config['ssl'] is still empty at this point then a ssl connection is not
+    # being requested. We remove config['ssl'] because passing the 'ssl' argument to
+    # MySQLdb, even if empty, will still make MySQLdb attempt to initiate a SSL connection
+    # (as per documentation). In the rare case that SSL support is disabled on
+    # the server completely this will raise an exception.
+    #
+    # The following prevents an exception being raised in the rare case where no SSL support is
+    # available in the client, and the user does not require an SSL connection anyway.
+    if config.get('ssl') == {}:
+        del config['ssl']
```

Comment:
not a good test for all cases, you might want this:

`if 'ssl' in config and not config['ssl']:`